error[E0521]: borrowed data escapes outside of closure
  --> tests/ui/thread_app_define_capture.rs:34:9
   |
32 |       fusible::kernel_enter(|app_define| {
   |                              ----------
   |                              |
   |                              `app_define` declared here, outside of the closure body
   |                              `app_define` is a reference that is only valid in the closure body
33 |           let app_define = UnsafeAppDefine(app_define);
34 | /         app_define
35 | |             .create_thread(
36 | |                 thread.into_ref(),
37 | |                 STACK.take().unwrap(),
...  |
41 | |                 },
42 | |             )
   | |_____________^ `app_define` escapes the closure body here
   |
   = note: requirement occurs because of the type `UnsafeAppDefine<'_, '_>`, which makes the generic argument `'_` invariant
   = note: the struct `UnsafeAppDefine<'ad, 'pke>` is invariant over the parameter `'ad`
   = help: see <https://doc.rust-lang.org/nomicon/subtyping.html> for more information about variance
